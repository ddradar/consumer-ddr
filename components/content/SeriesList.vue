<template>
  <ul>
    <li v-for="slug in series" :key="slug">
      <NuxtLink :to="`/series/${slug}/`">{{ seriesList[slug].title }}</NuxtLink>
    </li>
  </ul>
</template>

<script lang="ts" setup>
import seriesList from '~~/content/series/.seriesList.json'
import type { Song } from '~~/src/content'

const NuxtLink = resolveComponent('NuxtLink')

interface SeriesListProps {
  series: Song['series']
}

defineProps<SeriesListProps>()
</script>
