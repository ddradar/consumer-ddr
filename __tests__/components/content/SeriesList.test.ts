import { mount, RouterLinkStub } from '@vue/test-utils'
import { describe, expect, test, vi } from 'vitest'

import SeriesList from '~~/components/content/SeriesList.vue'

vi.mocked(resolveComponent).mockReturnValue(RouterLinkStub as any)

describe('components/content/SeriesList.vue', () => {
  test('renders series link', () => {
    const wrapper = mount(SeriesList, {
      props: { series: ['1st-jp', 'gb', 'us'] }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
