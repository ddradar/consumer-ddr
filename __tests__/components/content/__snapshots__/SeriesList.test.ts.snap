// Vitest Snapshot v1

exports[`components/content/SeriesList.vue > renders series link 1`] = `
<ul>
  
  <li>
    <a>
      Dance Dance Revolution (JP)
    </a>
  </li>
  <li>
    <a>
      Dance Dance Revolution GB
    </a>
  </li>
  <li>
    <a>
      Dance Dance Revolution (US)
    </a>
  </li>
  
</ul>
`;
